api-client/src/test/java/com/xing/api/CallSpecTest.java

```diff
@@ -1206,11 +1211,69 @@ public void canOverrideReadAndConnectTimeoutPerCall() throws Exception {
     }
 
     @Test
-    public void disallowZeroOrNegativeTimeouts() throws Exception {
+    public void canOverrideWriteTimeoutPerCall() throws Exception {
+        // for some unknown reason the timeout is not applied properly when run on Travis CI
+        // so we skip this test on the CI and only execute it locally
+        assumeThat(System.getenv("TRAVIS"), not(equalTo("true")));
```

Comment:
No, i think i believe `testSubscriber.assertError(SocketTimeoutException.class);` this part is already testing what you want to test and IMHO you don't need to calculate spent time and try to measure on a process might be vary on any machine.